pkg/costmodel/costmodel.go

```diff
@@ -1023,13 +1015,13 @@ func GetMemorySize(projectID, zone, machineType string, machineTypeMemoryMap map
 
 			memoryMb, err = getMachineType(projectID, zone, machineType, ctx)
 			if err != nil {
-				log.Warnf("Failed to get machine type: %s  err: %s",machineType, err.Error())
+				log.Warnf("Failed to get memoryinfo machine type: %s  err: %s",machineType, err.Error())
 				return 0, err
 			}
 		}
 		
 	} else {
-		memoryMb, exists = getMachineTypeFromMap(machineType, machineTypeMemoryMap)
+		memoryMb, exists = machineTypeMemoryMap[machineType]
 
 		if !exists {
 
```

Comment:
```
machineTypeMemoryMap[machineType] = memoryMb
```
This should be before the return statement and only if `machineType` doesn't exists in `machineTypeMemoryMap `

Reply:
ok